ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/ProhibitedEJBAnnotationsDeploymentUnitProcessor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2011, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.ejb3.deployment.processors;
+
+import org.jboss.as.ejb3.logging.EjbLogger;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.server.deployment.Attachments;
+import org.jboss.as.server.deployment.annotation.CompositeIndex;
+
+/**
+ * ProhibitedEJBAnnotationsDeploymentUnitProcessor class.
+ * Fail the deployment when it have EJB prohibited annotations
+ * <p/>
+ * @author Eduardo Sant'Ana da Silva
+ */
+public class ProhibitedEJBAnnotationsDeploymentUnitProcessor implements
+        DeploymentUnitProcessor {
+
+    @Override
+    public void deploy(DeploymentPhaseContext phaseContext)
+            throws DeploymentUnitProcessingException {
+        final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
+        final CompositeIndex index = deploymentUnit
+                .getAttachment(Attachments.COMPOSITE_ANNOTATION_INDEX);
+
+        for (final ProhibitedEJBAnnotations annotation : ProhibitedEJBAnnotations
+                .values()) {
+            if (!index.getAnnotations(annotation.getDotName()).isEmpty()) {
```

Comment:
You need to look at the EJB component descriptions, iterate through them and make sure that no component classes have this annotation. You current code looks at every class in the deployment, which makes it impossible to use the transaction interceptor.